ADR docs: justify a new CMS app for course to library import (#36380)
openedx/edx-platform: docs/decisions/0023-add-course-to-library-import-app.rst

Add new Django application for Modulestore to Learning Package import
=====================================================================

Status
------
Proposed

Context
-------

> As part of the Library Overhaul project, a new feature is required to
> handle a course content import process from the Modulestore into a
> Learning Core based Learning Package preserving the Import Log.
> This feature will enable users to populate Learning Package (today, that
> is always Content Libraries) with existing course structures and content.
> Currently, there is no mechanism to manage the Import process gradually
> preserving the Import Log for a further analyze.
> This creates a significant manual effort when users want to reuse existing
> course content within a library.

Reviewer: Some suggested edits for simplicity / grammar:
```suggestion
As part of the Libraries Overhaul project, a new feature is required to
handle the course content import process. This feature will enable
users to populate a Content Library with existing course structures
and content (from a Modulestore course into a Learning Core based
Library / Learning Package, preserving the details in an Import Log).
Currently, there is no mechanism to manage the Import process gradually
nor to create an Import Log for further analysis; this creates a significant
manual effort when users want to reuse existing course content within a library.
```

This ADR focuses specifically on creating a new Django application
*within the `cms` service* to handle this import functionality.
The application will, initially, *only* support importing from the
local Open edX platform's Modulestore.  Support for other sources
(e.g., remote LMSs) is explicitly out of scope for this initial
implementation but is a consideration for future evolution.

The application will provide the following functionality:

*   A Python API for programmatic access to the import functionality.

> *   A REST API for external interactions (e.g., from the frontend).

Reviewer: ```suggestion
*   A REST API for external interactions (e.g. from the frontend).
```
